models/search.php

```diff
@@ -3,8 +3,156 @@
  * Define the search class.
  */
 
+use TMS\Theme\Base\PostType\BlogArticle;
+use TMS\Theme\Base\PostType\Page;
+use TMS\Theme\Base\PostType\Post;
+use TMS\Theme\Base\Traits\Breadcrumbs;
+
 /**
  * The Search class.
  */
 class Search extends BaseModel {
+
+    use Breadcrumbs;
+
+    /**
+     * Get search results.
+     *
+     * @throws Exception If global $post is not available or $id param is not defined.
+     * @global WP_Query $wp_query The main query object.
+     */
+    public function results() {
+        global $wp_query;
+
+        $search_clause = get_search_query();
+        $result_count  = $wp_query->found_posts;
+        $results_text  = $wp_query->have_posts()
+            ? sprintf(
+            // translators: 1. placeholder is number of search results, 2. placeholder contains the search term(s).
+                _nx(
+                    '%1$1s result found for "%2$2s"',
+                    '%1$1s results found for "%2$2s"',
+                    $result_count,
+                    'search results summary',
+                    'tms-theme-base'
+                ),
+                $result_count,
+                $search_clause
+            )
+            : __( 'No search results', 'tms-theme-base' );
+
+        return [
+            'summary'    => $results_text,
+            'posts'      => $wp_query->have_posts() ? $this->enrich_results( $wp_query->posts ) : [],
+            'pagination' => [
+                'paged'          => $wp_query->query_vars['paged'] > 1 ? $wp_query->query_vars['paged'] : 1,
+                'posts_per_page' => $wp_query->query_vars['posts_per_page'],
+                'found_posts'    => $wp_query->found_posts,
+                'max_num_pages'  => $wp_query->max_num_pages,
+            ],
+        ];
+    }
+
+    /**
+     * Get template classes.
+     *
+     * @return array
+     */
+    public function template_classes() {
+        return apply_filters(
+            'tms/theme/search/search_item',
+            [
+                'search_item' => apply_filters( 'tms', 'has-background-secondary' ),
+            ]
+        );
+    }
+
+    /**
+     * Enrich results.
+     *
+     * @param array $posts Posts.
+     *
+     * @return mixed
+     */
+    private function enrich_results( $posts ) {
+        foreach ( $posts as $post_item ) {
+            switch ( $post_item->post_type ) {
+                case Page::SLUG:
+                    $post_item->content_type = get_post_type_object( Page::SLUG )->labels->singular_name;
+
+                    $meta = dustpress()->render( [
```

Comment:
Modelissa vain datan palauttaminen rendaus partiaalilla.
Vai onko tässä jokin erikoistapaus esim. javascriptillä rendaus? 🤔

Jos page loadissa rendaus, niin partiaalissa rendaus.

Eli tallenna postaukselle esim.
post->breadcrumbs postauksen juureen sitte postauksen kohdalla passaat tuon arvon breadcrumbs:lle

Ajatustasolla näin en nyt tutkinut tarkkaan miten noi partiaalien kansiorakenteet ja datat menee.
{>"breadcrumbs" data=post.breadcrumbs /}

Reply:
Ei erikoistapausta - ajatuksena tässä oli logiikan minimointi dustin päässä. Tästä taisi kuitenkin tulla lopputuloksena monimutkaisempi. Muutettu ehdotuksen mukaisesti! 👍